### ISSUE TITLE
Temporary blocking from Amazon server due too many requests while scanning?

### ISSUE THREAD

**Issue by playbackandrewind** [deburau/AmazonVineExplorer#4]
In the last days I often get a message from the Amazon Web server which says that there are technical issues:

"Tut uns Leid!
Während wir Ihre Eingabe ausführen wollten, ist ein technischer Fehler aufgetreten. Wir arbeiten bereits daran und werden sobald wie möglich wieder für Sie da sein. Bitte schauen Sie später wieder vorbei."

This never happened before, but after I switched to this version of AVE. Does anyone else get these error messages? I have the feeling that the message always appears when the scanner starts up. Of course, I could be imagining it, hence my question.

I set the scanning speed values both to 9999 instead of 6000, but that didn't help. 

It only affects https://www.amazon.de/vine/ - not the regular store https://www.amazon.de - this runs flawless.

**Comment by aemsx** [comment]
switching to a new external IP (by reconnect) or using another netork and then switching back solved this for me most of the time

**Comment by deburau** [comment]
Yes, I've had this error message several times, too. As aemsx commented, changing one's IP helps.
The message comes from the CDN, not from Amazon.
I think that I got the error when I browsed the vine items too fast while the background scan was still running. So maybe, just clicking more slowly might help.

**Comment by Olum-hack** [comment]
Hello @deburau,
this your last update I always get kicked out.
Is it possible that your scanner runs as fast as possible? the scanning speed has no influence.

**Comment by deburau** [comment]
What do you mean by "I always get kicked out"?

Here are the settings which are working perfectly for me:

![image](https://github.com/user-attachments/assets/99f40384-e083-4831-9955-dcfc64d09b2e)

**Comment by deburau** [comment]
See the reply from aemsx. Getting a new IP address normally helps. Or wait for some hours.

**Comment by deburau** [comment]
After weeks I'm having the issue again, without doing anything special. Another alternative to changing one's IP address is to use a VPN.

**Comment by playbackandrewind** [comment]
The only thing that was important for me is that my vine account gets not permanently banned. Hence my first question. So if the most of you having the same issue, it’s not a problem at all. If the web server blocks the requests because it is overloaded, I have fewer problems with that. But it would be really stupid if the account was lost because of a script error.

In my opinion thus is not a bug or similar, the issue should be closed asap.

Thx!

**Comment by Olum-hack** [comment]
I still get the blocking if I look at items with many variantes.

Now I deactivated some code and I don't get blocked. And I didn't find any negativ.
It is around line 2780+-

async function requestProductDetails(prod) {
    return new Promise(async (resolve, reject) => {
        if (prod.data_asin_is_parent) {// Lets get the Childs first
            fetch(`${window.location.origin}/vine/api/recommendations/${prod.id}`.replace(/#/g, '%23')).then(r => r.json()).then(async (res) => {
                if (res.error) {
                    if (res.error.exceptionType == 'ITEM_NOT_IN_ENROLLMENT') {
                        prod.forceRemove = true;
                        resolve(prod);
                    } else {
                        console.error('requestProductDetails():ERROR:', res.error);
                        reject(res.error.exceptionType);
                    }
                }
                const _data = res.result;
                console.log('DATA:', _data)
                prod.data_childs = _data.variations || [];
                const _promArray = new Array();
                prod.data_estimated_tax_prize = prod.data_estimated_tax_prize || 0;
// #OLUM kann problmem mit vielen Varianten verursachen und führt zum temprären Sperren.
//                 for (const _child of prod.data_childs) {
//                     _promArray.push(fetch(`${window.location.origin}/vine/api/recommendations/${(prod.id).replace(/#/g, '%23')}/item/${_child.asin}`.replace(/#/g, '%23')).then(r => r.json()).then((childData) => {
//                         console.log('CHILD_DATA:', childData);
//                         if (!childData.error) {

//                             // Copy over all returned datapoints od child asin
//                             for (const _datapoint of Object.keys(childData.result)) {
//                                 _child[_datapoint] = childData.result[_datapoint];
//                             }

//                             if (prod.data_estimated_tax_prize < _child.taxValue) {
//                                 prod.data_estimated_tax_prize = _child.taxValue;
//                                 prod.data_tax_currency = _child.taxCurrency;
//                             }
//                         }
//                     }))
//                 }
                Promise.all(_promArray).then((values) => {
                    console.log('All fetches returned: ', values);
                    resolve(prod);
                });
            })`

**Comment by deburau** [comment]
Thank you very much for resolving this annoying bug! I will publish a new version asap.

**Comment by playbackandrewind** [comment]
Many thanks to everyone involved in this great project!

**Comment by deburau** [comment]
Version 0.11.3 was released with Olum-hack's workaround for the issue.
